webapp/src/components/chat/message.tsx

```diff
@@ -214,6 +242,31 @@ export function Message({
 		</div>
 	);
 
+	const renderMessageContent = () => {
+		if (message === '⛔ MAX_RECURSION_LIMIT REACHED') {
+			return (
+				<div>
+					<span>
+						⛔ Conversation max limit reached, click{' '}
+						<button onClick={restartSession} className='text-blue-500 dark:text-blue-300'>
+							here
+						</button>{' '}
+						to restart the chat{' '}
+					</span>
+				</div>
+			);
+		}
```

Comment:
Instead, can you:

1. move this along with the restartsession api call to its own component ("ChatRestartMessage" or something), so we dont start polluting Message with special messages
2. create a mapping between the messages and the special component i.e { '⛔ MAX_RECURSION_LIMIT REACHED': ChatRestartMessage }
3. render it inside the useMemo of MessageBody and render it based on the map e.g. specialMessages['⛔ MAX_RECURSION_LIMIT REACHED']

That will make it cleaner and allows us to extend it for other custom components with different message texts (or even other properties of the message).

Reply:
ChatRestartMessage added.

Reply:
:ok_hand: